site/migrate-mcq-to-qq.md

```diff
@@ -0,0 +1,288 @@
+
+### Queue Length Limit overflow set to `reject-publish-dlx`
+
+The queue length exceeded with `overflow` set to [`reject-publish-dlx`](https://rabbitmq.com/maxlength.html#overflow-behaviour) is not supported by quorum queues. The `reject-publish-dlx` value is not supported.
+
+With mirrored classic queues, publishing to a full queue with `reject-publish-dlx` resulted in RabbitMQ republishing a rejected message to a dead letter exchange. With quorum queues, to apply the same logic, you must change `reject-publish-dlx` to `reject-publish`. Then, handle negative acknowledgements: after getting a negative acknowledgement, the application must publish the message again to a different exchange.
+
+To find out if  `overflow` set to `reject-publish-dlx` is configured for the mirrored classic queues you want to migrate, check for the `x-max-priority` string in the list of queues output that is provided by running the command in [Finding the Mirrored Classic Queues for Migration](#find-mcq) or you can also search for the ``reject-publish-dlx`` string in the source code.
+
+### Global QoS for Consumers
+
+Global [QoS prefetch](https://rabbitmq.com/quorum-queues.html#global-qos) where a channel sets a single prefetch limit for all consumers using that channel is not supported by quorum queues. If this functionality is required, try achieving the same results using alternative methods, for example, one solution might be to use a lower per-consumer QoS (given the known application load pattern).
+
+To find out if this feature is used, run the following command on a running system and check for non-empty output:
+
+```bash
+rabbitmqctl list_channels pid name global_prefetch_count | sed -n '/\t0$/!p'
+```
+
+A list of channel PIDs that have global QoS turned on are returned. Then, run the following command to map the channel PID to a queue name to verify if it is a mirrored classic queue. 
+
+```bash
+rabbitmqctl list_consumers queue_name channel_pid
+```
+### `x-cancel-on-ha-failover` for Consumers
+
+Mirrored queues consumers can be [automatically
+cancelled](https://www.rabbitmq.com/ha.html#cancellation) when a queue
+leader fails over. This can cause loss of information about which messages were sent to which consumer, and result in the same messages being sent again (duplicate messages). 
+
+Quorum queues will not have the same results in this situation i.e. duplicate messages are not sent again except when there is a node failure, or messages are resent for inflight messages when the consumer is cancelled or the channel is closed.
+
+In summary, with mirrored classic queues, you can observe possible duplicates in some cases. With quorum queues, this observation is not possible but duplicates messages can still happen (less frequently) for some of the reasons they happened when using mirrored classic queues. 
```

Comment:
I think that first sentence says that "observation" = "duplicates". Then the second on is contradictory, saying that "observation is not possible", "duplicates can still happen".

Some more emphasis is needed on the fact that application still needs to be prepared for duplicates, something along the lines:

With CMQs you can observe possible duplicates in some cases. Some of these cases were covered by `x-cancel-on-ha-failover`, and others weren't. Most of the cases covered by `x-cancel-on-ha-failover` went away with QQs. But the ones that aren't covered are still here, and your application needs to be aware / to be able to handle duplicates anyway.

Reply:
Updates made for this comment, please review again @binarin